crates/node/src/command.rs

```diff
@@ -52,69 +52,138 @@ impl SubstrateCli for Cli {
+        local_path,
+    ) {
+        (Some(chain_spec_url), _, _, _, _) => {
+            utils::fetch_from_url(
+                chain_spec_url.clone(),
+                madara_path.clone() + "/chain-specs",
+                cli.run.update_configs,
+            )?;
+            let chain_spec = chain_spec_url.split('/').last().expect("Chain spec file name not found");
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/" + chain_spec);
+        }
+        (None, Some(Testnet::Sharingan), false, false, Ok(ref src_path)) => {
+            let src_path = src_path.clone() + "/configs/chain-specs/testnet-sharingan-raw.json";
+            utils::copy_from_filesystem(src_path, madara_path.clone() + "/chain-specs", cli.run.update_configs)?;
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
+        }
+        (None, Some(Testnet::Sharingan), false, false, _) => {
+            let madara_configs: configs::Configs =
+                serde_json::from_str(&utils::read_file_to_string(madara_path.clone() + "/configs/index.json")?)
+                    .expect("Failed to load madara configs");
+            // If testnet flag is specified, right now it's only Sharingan
+            for chain_spec in madara_configs.chain_specs {
+                configs::fetch_and_validate_file(
+                    madara_configs.remote_base_path.clone(),
+                    chain_spec,
+                    madara_path.clone() + "/configs/chain-specs/",
+                    cli.run.update_configs,
                 )?;
             }
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         }
+        _ => {}
     }
 
-    // TODO confirm with the CI that we are fetching all and fetch dynamically
-    // Issue #982
-    for file in constants::CAIRO_CONTRACTS_FILES {
-        let src_path = utils::get_project_path();
-        if let Ok(src_path) = src_path {
-            let src_path = src_path + "/configs/cairo-contracts/" + file;
-            utils::copy_from_filesystem(src_path, madara_path.clone() + "/cairo-contracts")?;
-        } else {
-            utils::fetch_from_url(
-                constants::CAIRO_CONTRACTS_URL.to_string() + file,
-                madara_path.clone() + "/cairo-contracts",
+    if cli.run.run_cmd.shared_params.chain.is_some() {
+        cli.run.run_cmd.rpc_external = true;
+        cli.run.run_cmd.rpc_methods = RpcMethods::Unsafe;
+    }
+
+    Ok(())
+}
+
+fn fetch_madara_configs(cli: &mut Cli) -> Result<(), String> {
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
+    let local_path = utils::get_project_path();
+
+    if let Some(configs_url) = cli.run.configs_url.clone() {
+        utils::fetch_from_url(configs_url, madara_path.clone() + "/configs", cli.run.update_configs)?;
```

Comment:
the fn name `utils::fetch_from_url` imply (to me) that the fn will return the fetched values that I can then use to do whatever I want.

This function instead write the fetched values directly to the disk. A more descriptive name is needed. Maybe "write_madara_confings_to_disk_from_url".

Is it madara_configs or chain_specs that are fetched?

Reply:
both madara_configs and chain_specs use that function to fetch files from the URL and add them in the madara path

Reply:
so maybe just `fetch_from_url_and_write_to_disk`